command/collector.go

```diff
@@ -0,0 +1,47 @@
+package command
+
+import (
+	"errors"
+	"fmt"
+	"github.com/bwmarrin/discordgo"
+	"strconv"
+)
+
+// MessageCollector
+// /collect <message-snowflake-id> <condition-type> <condition-value>
+// Condition type: number, time
+// Condition value: # Messages to collect starting with ID or timestamp to collect till
+type MessageCollector struct{}
+
+
+func (h *MessageCollector) Name() string {
+	return "Message Collector"
+}
+
+func (h *MessageCollector) Description() string {
+	return "Collects messages starting from a given message ID to a given limit to local storage for further processing"
+}
+
+func (h *MessageCollector) Handle(s *discordgo.Session, m *discordgo.MessageCreate, args []string) (processedMessage bool, err error) {
+	if args[0] != "/collect" {
+		return
+	}
+
+	processedMessage = true
+
+	if len(args) != 3 {
+		err = errors.New("invalid arguments")
+		return
+	}
+
+	numMessagesToCollect, err := strconv.ParseUint(args[2], 10, 32)
+
+	if err != nil {
+		err = fmt.Errorf("unable to parse value for number of messages to collect: %v", err)
+		return
+	}
```

Comment:
`err` is already declared (since it is a named return variable), but here it's being redeclared (because of the `:=`). This is called shadowing and can lead to [all sorts of problems when scoped](https://groups.google.com/g/golang-nuts/c/VeJTB9dhhK4). In this particular case, since the scope is the parent scope, this would technically work, but shadowing variables is generally considered bad practice. Therefore, I suggest one of these approaches (preferably the first):
```suggestion
	var numMessagesToCollect uint64
	if numMessagesToCollect, err = strconv.ParseUint(args[2], 10, 32); err != nil {
		err = fmt.Errorf("unable to parse value for number of messages to collect: %v", err)
		return
	}
```
or
```suggestion
	numMessagesToCollect, err1 := strconv.ParseUint(args[2], 10, 32)
	if err1 != nil {
		err = fmt.Errorf("unable to parse value for number of messages to collect: %v", err1)
		return
	}
```

Reply:
oops